pkg/solana/solana_service.go

```
@@ -0,0 +1,748 @@
+		EventIdl:        idl,
+		SubkeyPaths:     logpollertypes.SubKeyPaths(f.SubkeyPaths),
+		Retention:       f.Retention,
+		MaxLogsKept:     f.MaxLogsKept,
+		IncludeReverted: f.IncludeReverted,
+	}, nil
+}
+
+func convertAccounts(accs []*rpc.Account) []*commonsol.Account {
+	ret := make([]*commonsol.Account, 0, len(accs))
+	for _, acc := range accs {
+		ret = append(ret, &commonsol.Account{
+			Lamports:   acc.Lamports,
+			Owner:      commonsol.PublicKey(acc.Owner),
+			Data:       convertDataBytesOrJSON(acc.Data, ""),
+			Executable: acc.Executable,
+			RentEpoch:  acc.RentEpoch,
+			Space:      acc.Space,
+		})
+	}
+
+	return ret
+}
+
+func convertAccountResult(acc *rpc.GetAccountInfoResult, enc commonsol.EncodingType) *commonsol.GetAccountInfoReply {
+	if acc == nil {
+		return nil
+	}
+
+	var a *commonsol.Account
+	if acc.Value != nil {
+		a = &commonsol.Account{
+			Lamports:   acc.Value.Lamports,
+			Executable: acc.Value.Executable,
+			Owner:      commonsol.PublicKey(acc.Value.Owner),
+			Data:       convertDataBytesOrJSON(acc.Value.Data, enc),
+		}
+	}
+
+	return &commonsol.GetAccountInfoReply{
+		RPCContext: commonsol.RPCContext{
+			Context: commonsol.Context{
+				Slot: acc.Context.Slot,
+			},
+		},
+		Value: a,
+	}
+}
+
+func convertAccountInfoOpts(opts *commonsol.GetAccountInfoOpts) *rpc.GetAccountInfoOpts {
+	var ds *rpc.DataSlice
+	if opts.DataSlice != nil {
+		ds = &rpc.DataSlice{}
+		ds.Length = opts.DataSlice.Length
+		ds.Offset = opts.DataSlice.Offset
+	}
+
+	return &rpc.GetAccountInfoOpts{
+		Encoding:       solana.EncodingType(opts.Encoding),
+		Commitment:     rpc.CommitmentType(opts.Commitment),
+		MinContextSlot: opts.MinContextSlot,
+		DataSlice:      ds,
+	}
+}
+
+func convertDataBytesOrJSON(obj *rpc.DataBytesOrJSON, enc commonsol.EncodingType) *commonsol.DataBytesOrJSON {
+	if obj == nil {
+		return nil
+	}
+	if enc == "" {
+		enc = commonsol.EncodingJSON // default
+	}
+	var txJSON []byte
+	if b, err := json.Marshal(obj); err == nil {
```

err not being handled here ?

---

yeah you are right, if the caller called for json we should either return an error or fallback to binary. Pushing fixes now